closure/goog/i18n/numberformat.js

```diff
@@ -580,6 +588,173 @@ goog.i18n.NumberFormat.prototype.roundNumber_ = function(number) {
 
 
 /**
+ * Formats a number with the appropriate groupings when there are repeating
+ * digits present. Repeating digits exists when the length of the digits left
+ * of the decimal place exceeds the number of non-repeating digits.
+ *
+ * Formats a number by iterating through the integer number (intPart) from the
+ * most left of the decimal place by inserting the appropriate number grouping
+ * separator for the repeating digits until all of the repeating digits is
+ * iterated. Then iterate through the non-repeating digits by inserting the
+ * appropriate number grouping separator until all the non-repeating digits
+ * is iterated through.
+ *
+ * In the number grouping concept, anything left of the decimal
+ * place is followed by non-repeating digits and then repeating digits. If the
+ * pattern is #,##,###, then we first (from the left of the decimal place) have
+ * a non-repeating digit of size 3 followed by repeating digits of size 2
+ * separated by a thousand separator. If the length of the digits are six or
+ * more, there may be repeating digits required. For example, the value of
+ * 12345678 would format as 1,23,45,678 where the repeating digit is length 2.
+ *
+ * @param {Array<string>} parts The original parts variable that built in
+ *  goog.i18n.NumberFormat.prototype.subformatFixed_ used to build the
+ *  formatted number.
+ * @param {number} zeroCode The original zeroCode variable from
+ *  goog.i18n.NumberFormat.prototype.subformatFixed_ of displaying the
+ *  zero digit.
+ * @param {string} intPart The original intPart variable from
+ *  goog.i18n.NumberFormat.prototype.subformatFixed_ describing the integer
+ *  portion of the value.
+ * @param {Array<number>} groupingArray The array of numbers to determine the
+ *  grouping of repeated and non-repeated digits.
+ * @param {number} repeatedDigitLen The length of the repeated digits left of
+ *  the non-repeating digits left of the decimal.
+ * @return {!Array<string>} Returns the resulting parts variable containing
+ *  how numbers are to be grouped and appear.
+ * @private
+ */
+goog.i18n.NumberFormat.formatNumberGroupingRepeatingDigitsParts_ =
+    function(parts, zeroCode, intPart, groupingArray, repeatedDigitLen) {
+  // Keep track of how much has been completed on the non repeated groups
+  var nonRepeatedGroupCompleteCount = 0;
+  var currentGroupSizeIndex = 0;
+  var currentGroupSize = 0;
+
+  var grouping = goog.i18n.NumberFormatSymbols.GROUP_SEP;
+  var digitLen = intPart.length;
+
+  if (parts === null) {
```

Comment:
Wow, I just noticed `parts` passed to `subformatFixed_` is, incorrectly, non-nullable. That's why this is causing confusion. 

Can you add move this check to the top of `subformatFixed_` instead?